### PR TITLE
feat: form requirement admins should be able to create new forms

### PR DESCRIPTION
**Pull request by FBalint**

### CODE REVIEW

**Comment by FBalint** [comment]
Things that are not settled yet:

- the loading state on the dropdown is not working, there seems to be a state after isValidating, that I cannot track
- I put a `res?: any` type in many of the onSuccess callbacks, should we type that somehow?

**Review by BrickheadJohnny — COMMENTED** [review]
Left a couple of minor comments & also added types in 3b736f913c219f42bb02a0629b99d41c8c26ce8e, let me know if you like this solution!

File: src/requirements/Form/FormForm.tsx [hunk 1]
```diff
@@ -30,22 +33,33 @@ const FormForm = ({ baseFieldPath }: RequirementFormProps) => {
         value: form.id,
       })) ?? []
 
+  const handleFormAdded = (formId: number) => {
+    mutate().then(() => {
```
Comment (BrickheadJohnny) [inline src/requirements/Form/FormForm.tsx:37]: I think we don't need to call `mutate` here, we already mutate the cache directly inside `useCreateForm` ;)

File: src/requirements/Form/AddFormButton.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  const handleSuccess = (res: any) => {
+    const formId = res?.platformGuildData?.formId
+    return onSuccess(formId)
+  }
+
+  return (
+    <>
+      <FormProvider {...methods}>
+        <Button size="xs" variant="ghost" borderRadius={"lg"} onClick={handleOpen}>
+          <Text colorScheme={"gray"}>Create new</Text>
```
[the remark this answers is not in the document]
Comment (FBalint) [reply src/requirements/Form/AddFormButton.tsx:42]: Thanks, this was copied, I replaced it at the other place as well

**Review by BrickheadJohnny — APPROVED** [review]

**Pull request merged**